include/falcosecurity/internal/plugin_mixin_async.h

```diff
@@ -142,8 +142,7 @@ template<class Plugin, class Base> class plugin_mixin_async : public Base
                             const ss_plugin_async_event_handler_t h)
     {
         FALCOSECURITY_CATCH_ALL(Base::m_last_err_storage, {
-            auto handler = std::make_unique<async_event_handler>(o, h);
-            _dump_state(std::move(handler), static_cast<Plugin*>(this));
+            _dump_state(o, h, static_cast<Plugin*>(this));
```

Comment:
Basically, we cannot call `_dump_state` with a non-trivial C++ type (unique_ptr in the old case) because we might compile the variadic version of  `_dump_state` that does not accept the unique_ptr; see for example this redis-cpp issue: https://github.com/tdv/redis-cpp/issues/12 and its fix: https://github.com/tdv/redis-cpp/commit/4ed39c88916f2d3cb8d970de76c96fb7950b786c